### ISSUE TITLE
l3graphics, l3backend-graphics: allow additional attributes 

### ISSUE THREAD

**Issue by u-fischer** [latex3/latex3#1191]
In a tagging structure graphics (XObjects) can be referenced either similar to text with a MC-tag, but also (similar to link annotations) with an OBJR object. 

The second methods requires a /StructParent entry in the attributes of the graphic, to be able to reference it from the parent tree. 

It isn't yet clear if there are benefits if an OBJR is used, but to allow experiments it would be nice if the backends could be extended to allow to inject additional attributes.  E.g.  for pdftex/luatex:

~~~~
 attr ~
            {
             % user attribute  ...
              \tl_if_empty:NF \l_@@_decodearray_str
                { /Decode~[ \l_@@_decodearray_str ] }
              \bool_if:NT \l_@@_interpolate_bool
                { /Interpolate~true }
            }
~~~~

**Comment by josephwright** [comment]
Although it's had no real feedback, I assume that everyone is happy enough with the idea that info is passed to the `l3graphics` backend using keyvals. So we could simply add another keyval to the known set, and have that then passed as an internal variable to the backend. I guess then the question is are these generic or are they PDF-specific - i.e. do we need to worry about manipulating them at all if they come in for PostScript/SVG?

Assuming they are PDF-specific (I suspect so), they could be given as a `user-attr`/`custom`/`objr` key - any preferences?

**Comment by u-fischer** [comment]
My usecase is rather PDF specific and I think there is no interface for dvips for such attributes. I would ignore that backend (and SVG too) and simply document that the key does nothing there. `user-attr` or `pdf-attr` sounds ok, `objr` is imho to specific.

**Comment by josephwright** [comment]
Speaking of `l3graphics`, if we really do want to use this for tagging, is it worth raising the question of moving to `l3kernel`?

**Comment by josephwright** [comment]
Fixed by c5340639a5dd7b73d410fecc297f0ff6bb4b7f4a
